### PR TITLE
kernel: add confidential guest build option

### PR DESCRIPTION
**Pull request by jimcadden**
Includes support for SEV guest kernels

### CODE REVIEW

File: tools/packaging/kernel/build-kernel.sh [hunk 1]
```diff
@@ -98,6 +100,7 @@ Options:
 	-p <path>   	: Path to a directory with patches to apply to kernel.
 	-t <hypervisor>	: Hypervisor_target.
 	-v <version>	: Kernel version to use if kernel path not provided.
+	-x <type>	: Confidential guest protection type, such as sev/tdx/pef
```
Comment (bergwolf) [inline tools/packaging/kernel/build-kernel.sh:103]: please just list `sev` here as the PR only adds support for `sev` configs.

File: tools/packaging/kernel/build-kernel.sh [hunk 2]
```diff
@@ -461,6 +474,13 @@ main() {
 			v)
 				kernel_version="${OPTARG}"
 				;;
+			x)
+				conf_guest="${OPTARG}"
+	      case "$conf_guest" in
+		      sev) ;;
```
Comment (devimc) [inline tools/packaging/kernel/build-kernel.sh:480]: it'd be great if you can add a log, ie: " Enabling confidential guest protection: SEV"

**Review by devimc — COMMENTED** [review]
thanks @jimcadden

File: tools/packaging/kernel/build-kernel.sh (hunk 2 above)
Comment (jimcadden) [reply tools/packaging/kernel/build-kernel.sh:480]: There is a similar log comment already added: 
```
215   if [[ "${conf_guest}" != "" ]];then$
216      info "Add kernel config for ${conf_guest} guest protection"$
```

**Comment by jimcadden** [comment]
@bergwolf @devimc Thanks for the comments. All set for now.

**Review by devimc — APPROVED** [review]
thanks  @jimcadden

**Comment by bpradipt** [comment]
@jimcadden few minor nits and the CI will be happy.  The commit message should include a `Fixes` entry - https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format

**Review by bergwolf — APPROVED** [review]
Thanks!

**Pull request merged**